lib/utils.ts

```diff
@@ -59,10 +59,36 @@ export function generateBuildTasks(
 				img.image.dockerfile != null
 					? { dockerfilePath: img.image.dockerfile }
 					: {},
+				// Pass through args, context, labels, tag.
+				//
 				// It's possible to specify an image name as well as
 				// a build, but that doesn't make sense in a balena
 				// ecosystem, so we remove it
-				_.omit(img.image, 'image'),
+				//
+				// We also remove anything that goes into dockerOpts in the next step
+				_.omit(
+					img.image,
+					'image',
+					'cache_from',
+					'extra_hosts',
+					'network',
+					'shm_size',
+					'target',
+				),
+				// pass through build options from composition
+				// translating to dockerode ImageBuildOptions properties
+				{
+					dockerOpts: _.merge(
+						{},
+						img.image.cache_from ? { cachefrom: img.image.cache_from } : {},
+						img.image.network ? { networkmode: img.image.network } : {},
```

Comment:
let's continue the conversation from https://github.com/balena-io-modules/resin-compose-parse/pull/59#pullrequestreview-733128487